src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1014,10 +988,10 @@ public class ForkJoinPool extends AbstractExecutorService {
     static final int INVALID_ID       = 0x4000;   // unused external queue id
 
     // pool.runState bits
-    static final int STOP             = 1 <<  0;   // terminating
-    static final int SHUTDOWN         = 1 <<  1;   // terminate when quiescent
-    static final int TERMINATED       = 1 <<  2;   // only set if STOP also set
-    static final int RS_LOCK          = 1 <<  3;   // lowest seqlock bit
+    static final long STOP            = 1L <<  0;   // terminating
+    static final long SHUTDOWN        = 1L <<  1;   // terminate when quiescent
+    static final long TERMINATED      = 1L <<  2;   // only set if STOP also set
+    static final long RS_LOCK         = 1L <<  3;   // lowest seqlock bit
```

Comment:
@DougLea I presume making these longs was to enable the int-field padding?

Reply:
The net impact is to place runState field nearer queues ref, which on average reduces cacheline accesses since the are almost always accessed together. (The main focus of this update is to reduce memory contention that would otherwise accompany greater utilization. Most of the adjustments are small, but add up.)